rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala

```diff
@@ -130,44 +113,37 @@ class ParVisitSpec extends FlatSpec with Matchers {
         new PVar("x"),
```

Comment:
@guardbotmk3 Could you explain this test? Does this mean that

`for (x <- y) P | for (x <- z) Q`

is invalid if `x` is not bound at the time something hits either channel?

Reply:
At the top level it is an error for there to be any free variables.

But we don't know we're at the top level as we're normalizing, so that is checked afterwords

It's an error for a pattern to have a free variable used more than once, because what do you bind it to, or do you enforce that they're structurally equivalent?